client/src/components/Chat.tsx

```diff
@@ -0,0 +1,267 @@
+  padding: 0px 4px;
+
+  p {
+    margin: 3px;
+    text-shadow: 0.3px 0.3px black;
+    font-size: 15px;
+    font-weight: bold;
+    line-height: 1.4;
+    overflow-wrap: anywhere;
+  }
+
+  :hover {
+    background: #3a3a3a;
+  }
+`
+
+const InputWrapper = styled.div`
+  box-shadow: 10px 10px 10px #00000018;
+  border: 1px solid #42eacb;
+  border-radius: 0px 0px 10px 10px;
+  display: flex;
+  flex-direction: row;
+  background: linear-gradient(180deg, #000000c1, #242424c0);
+`
+
+const InputTextField = styled(InputBase)`
+  border-radius: 0px 0px 10px 10px;
+  input {
+    padding: 5px;
+    font-size: 15;
+  }
+`
+
+const colorArr = ['#7bf1a8', '#ff7e50', '#9acd32', '#daa520', '#ff69b4', '#c085f6']
+
+// determine name color by first character charCode
+const getColorByName = (author: string) => {
+  return colorArr[Math.floor(author.charCodeAt(0) % colorArr.length)]
+}
+
+const dateFormatter = new Intl.DateTimeFormat('en', {
+  timeStyle: 'short',
+  dateStyle: 'short',
+})
+
+export default function Chat() {
+  const [inputValue, setInputValue] = useState('')
+  const [showChat, setShowChat] = useState(true)
+  const [showEmojiPicker, setShowEmojiPicker] = useState(false)
+  const messagesEndRef = useRef<HTMLDivElement>(null)
+  const chatMessages = useAppSelector((state) => state.chat.chatMessages)
+  const game = phaserGame.scene.keys.game as Game
+
+  const handleChange = (event: React.FormEvent) => {
+    var input: any = event.target
+    setInputValue(input.value)
+  }
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      setInputValue('')
+    } else if (event.key === 'Enter') {
+      handleSubmit()
+    }
+  }
+
+  const handleSubmit = () => {
+    var val = inputValue.trim()
+    if (val) {
+      setInputValue('')
+      game.network.addChatMessage(val)
+      game.myPlayer.updateDialogBubble(val)
+    }
+  }
+
+  const scrollToBottom = () => {
+    messagesEndRef.current?.scrollIntoView({ behavior: 'smooth' })
+  }
+
+  const Message = ({ chatMessage, messageType }) => {
+    const [tooltipOpen, setTooltipOpen] = useState(false)
+
+    return (
+      <MessageWrapper
+        onMouseEnter={() => {
+          setTooltipOpen(true)
+        }}
+        onMouseLeave={() => {
+          setTooltipOpen(false)
+        }}
+      >
+        <Tooltip
+          open={tooltipOpen}
+          title={dateFormatter.format(chatMessage.createdAt)}
+          placement="right"
+          arrow
+        >
+          {messageType === MessageType.REGULAR_MESSAGE ? (
+            <p
+              style={{
+                color: getColorByName(chatMessage.author),
+              }}
+            >
+              {chatMessage.author}{' '}
+              <span style={{ color: 'white', fontWeight: 'normal' }}>{chatMessage.content}</span>
+            </p>
+          ) : (
+            <p style={{ color: 'grey', fontWeight: 'normal' }}>
+              {`${chatMessage.author} ${chatMessage.content}`}
```

Comment:
this string interpolation seems redundant

`{chatMessage.author} {chatMessage.content}` will do